### DISCUSSION TITLE
How to add Project Reference to the library projects in the Mobile project ?

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#9442] [Q&A]
[_From Discord_:](https://discord.com/channels/372137812037730304/550416151172087808/1006203467821494443)

I'm working on migrating my Uno-Platform app that is targeting iOS and Android from Core 3.1/Standard2.0 to Net6.
Previously I was able to use Project References that point to my C# library projects, e.g. <myapp>.Library, <myapp>.Data.Models, etc.
When I attempt to do this with a NET6 project adding a Project Reference to the library projects in the Mobile project, it breaks the XAML Code Gen (even with a completely empty library project, just referencing it). This results in all of my xaml.cs files unable to reference a gen'd partial, therefore InitializeComponent does not exist, etc.
Is it possible to have this clean separation of project files? Or, am I required to jam everything into the Shared Project?

**Comment by jeromelaban** [comment]
It's generally supported, by you may be missing something in your project dependencies that causes this issue. If you can create a bare minimum repro of what you're trying to do, it'll help to guide you.

**Reply by jspeaker** [reply]
I think I've found my own answer to this. It appears that any project references in the Mobile project have to be to Shared Projects, and that it is possible to create more than 1 shared project so that library code can be separated out along conceptual lines. I was attempting to use NET6 class libraries rather than Shared Projects.

**Reply by jeromelaban** [reply]
Thanks. Know that it's possible to use net6 for class libraries, but if you include XAML in those libraries, you'll need to use the uno "multi-platform library" template from Uno.

**Reply by jspeaker** [reply]
Thanks, Jerome. Looks like the way my solution/projects are set up that I do need to use the multi-platform library template. Looks the like the correct target frameworks are the primary "secret sauce" that a basic net6 library is missing. Hopefully I'm on my way at this point. Cheers.
